csi/server/plugin/opensds/controller.go

```diff
@@ -557,15 +559,124 @@ func (p *Plugin) ListSnapshots(
 	glog.V(5).Infof("start to ListSnapshots, MaxEntries: %v, StartingToken: %v, SourceVolumeId: %v, SnapshotId: %v!",
 		req.MaxEntries, req.StartingToken, req.SourceVolumeId, req.SnapshotId)
 
-	var opts = map[string]string{"Id": req.SnapshotId, "VolumeId": req.SourceVolumeId}
-	snapshots, err := Client.ListVolumeSnapshots(opts)
-
+	var opts map[string]string
+	allSnapshots, err := Client.ListVolumeSnapshots(opts)
 	if nil != err {
 		return nil, err
 	}
 
+	if len(allSnapshots) <= 0 {
+		glog.V(5).Info("len(allSnapshots) <= 0")
+		return &csi.ListSnapshotsResponse{}, nil
```

Comment:
If 0 != snapshotIDLen or 0 != sourceVolumeIdLen, should return NotFound.  Otherwise it is not consistent with the error handling below.